tests/test_server.py

```diff
@@ -162,6 +168,16 @@ def test_modifying_interceptor(aio):
     ) as client:
         assert client.Execute(DummyRequest(input="test")).output == "TEST"
 
+def test_modifying_interceptor():
```

Comment:
It looks like you copied the test above without changing the name, so this ends up replacing the previous test. It'll need a unique name.

Could you also add an async version for consistency with other tests?